spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1711,12 +1743,16 @@ protected void doInTransactionWithoutResult(TransactionStatus status) {
 							TransactionSupport.clearTransactionIdSuffix();
 						}
 					}
-					else if (KafkaMessageListenerContainer.this.getContainerProperties().isSyncCommits()) {
-						ListenerConsumer.this.consumer.commitSync(offsetsToCommit);
-					}
 					else {
-						ListenerConsumer.this.consumer.commitAsync(offsetsToCommit,
-								KafkaMessageListenerContainer.this.getContainerProperties().getCommitCallback());
+						ContainerProperties containerProps = KafkaMessageListenerContainer.this.getContainerProperties();
+						if (containerProps.isSyncCommits()) {
+							ListenerConsumer.this.consumer.commitSync(offsetsToCommit,
+									containerProps.getSyncCommitTimeout());
```

Comment:
Why don't you call `determineSyncCommitTimeout()` from here?

Reply:
Because I set the property if it was null.

Reply:
Because I set the property if it was null - see line 565.